### ISSUE TITLE
Consider integrating CiviCRM for registrations

### ISSUE THREAD

**Issue by laryn** [backdrop-ops/events.backdropcms.org#14]
For consideration, there has been discussion of a CiviCRM installation -- in which case we should consider using CiviCRM for event registrations.

Some form of support in implementing has been voiced on Zulip by @ericfg and @laryn

**Comment by jackaponte** [comment]
YES to this! I'm looping in @scoobird who has started doing initial work on a Civi implementation for Backdrop already!

**Comment by scoobird** [comment]
Hi, 
We had some conversations over a year ago about installing Civi into Backdropcms.org for contact management and providing support -- there weren't any technical issues with that, just fell to the bottom of my plate unfortunately. However I wasn't aware of this standalone events site at the time, so some further coordination sounds like a next step. I'll pop into Zulip.

**Comment by stpaultim** [comment]
> However I wasn't aware of this standalone events site at the time

@scoobird - Neither were we. :-) 

The events site is very new, it's only been active a few weeks. We set it up for Backdrop LIVE, but with the intention of using it for future events including Backdrop LIVE, Virtual User Groups, and other events. 

We were not thinking about CiviCRM for event registration at the time. But, that is a very interesting possibility. However, we also want CiviCRM for general contact management. I assume that we'll want to keep this all together.

So the events site was set up without any intention of integrating, other than a general desire we have to eventually create a single sign-on option for all the Backdrop sites. 

I don't think it's too late to rethink the entire idea for events.backdropcms.org. If we are going to use CiviCRM for event registration AND contact management on Backdropcms.org, maybe they should all be a single site?

I'm not sure what is possible right now. Maybe we should have a meeting to talk about our goals and discuss strategies?

**Comment by jackaponte** [comment]
Thanks for these reflections, @stpaultim! 

As Morgan mentioned, we started some initial work towards this but it's been stalled for a while. A meeting to discuss goals and strategies might be a good next step, but first let me share a little more about conversations to date.

Back in November 2019 @jenlampton, @scoobird and I had a meeting about getting CiviCRM set up for the Backdrop community for a few purposes:

- General contact management
- Mailings and newsletters (incl mailing list management)
- Donor and donation tracking
- Event registration

Keeping all of that data and functionality in one spot makes a lot of sense to me; the same people who are on the mailing list are also likely to attend events and make donations to the project, and having all of that information aggregated and correlated so you can better understand and connect with your community is a big reason why a fully-featured CRM like Civi is desirable. 

So I like the idea of keeping CiviCRM within backdropcms.org *unless* there are overwhelming privacy and security concerns about having all that CRM data inside of the main public-facing website where anyone can create their own account. We'll need to be extra-careful about user roles and permissions and about promptly applying CiviCRM security updates when they are released, which is no small task!

Folks have expressed interest in having CiviCRM in place in time to use it for event registration for the next Backdrop LIVE in July. If we want to open registration more than a month in advance, that means having Civi up and running by May.

I see two potential paths forward with this at this point:

1. We could proceed with earlier plans to get CiviCRM set up within the main BackdropCMS.org site. We can focus on getting the basics in place so that we're ready for event registration for the next Backdrop LIVE. After getting everything needed for event registration set up, including outgoing mail and potentially payment processing, we can look to moving other functionality and data into CiviCRM, namely mailings and newsletters.
2. We could first step back and have a meeting with all of us who are invested in getting Civi set up for Backdrop --@cellear and @larsdesigns come to mind!-- to get into better alignment about goals and strategies, including about where to house the Civi installation and whether to keep using events.backdropcms.org. Then we could move forward with the actual implementation.

Approach 1 requires less logistics up front and would probably allow us to get Civi up and running sooner. I'd endorse Approach 1, but Approach 2 is best if we find that we need better clarity and agreement about our goals and strategies here before we move on to technical implementation.

Either way, using Civi for the Backdrop community will take a fair amount of ongoing work and responsibility. That's both on the technical side of things (keeping Civi and its extensions up to date, troubleshooting the many problems that can come with Civi, configuration adjustments, etc) AND the data and constituent relationship management work needed in order to actually get the most out of having a CRM--actually setting up the event registrations, sending the mailings, thanking the donors and whatnot! All of that will certainly take a fair amount of planning and coordination in the future.

So, given all of that--what do folks think about next steps here? Shall we forge on with Approach 1? Step back for Approach 2? Something else?

**Comment by larsdesigns** [comment]
I would be up for option 2 to gain a solid plan.

**Comment by cellear** [comment]
I'm not in favor of stepping back, even when the other branch involves more of my involvement!  In fact, that even recommends it a little to me.  However, I'm certainly very interested in seeing this succeed, and happy to add anything useful that I can.

Perhaps it would be easier to bring up CiviCRM on events.backdropcms.org as a standalone project, and the integrate it with the "main" site once we're comfortable we understand the issues?  I would speculate (without being sure) that the integration points between events.backdropcms.org would be pretty similar to www.

**Comment by jenlampton** [comment]
I'd also be up for option #2, I wouldn't want to rush into things and end up making more work for ourselves later on. 

> and whether to keep using events.backdropcms.org

I don't think there's any question that we'll want to keep using our events system for posting presentations, scheduling discissions, etc, but we would like all the **registration data** collected there to get back into civi (likely on b.org). Having different properties on different subdomains (and even different servers) wasn't a problem for Drupal's civi instance back in 2010 (last time I used civi...) so I'd been imagining it wouldn't matter what we collected from which site... but this is also making me more convinced we should go with option #2, as there may be nuances I don't understand yet.

**Comment by oadaeh** [comment]
I think that regardless of the path forward, because things have changed in the > 1 year since the initial implementation was started, having a meeting with all the stakeholders to make sure everyone knows the current situation and current goals is the best idea, especially if making an initial stab at something as big as this might end up with the possibility that everything might need to be changed in the future. That seems like the potential for a lot of wasted effort.
Also, while I'm sure I don't get a vote on this and others might disagree, I think missing getting CiviCRM installed by the next Backdrop LIVE event with the idea of getting it set up correctly to begin with is totally acceptable.

**Comment by ericfg** [comment]
if option 2 is selected I would love to be involved in the process. As long as I'm not driving things, I have time to help with the planning and implementation of CiviCRM -- and long term maintenance/upkeep as well.

**Comment by stpaultim** [comment]
For me the priority for CiviCRM is contact management and volunteer management. I think we really need something like CiviCRM for that purpose and I would make that first priority.

For event management, we really can get by without CiviCRM for now. CiviCRM could be helpful and it might be nice to have event registration tied to our CRM, but I just don't see that as critical or as important as the contact management stuff. I think that are pretty close to a workable event registration system without CiviCRM and we I expect that we can add event management to CiviCRM at any point (if and when it makes sense). 

My main argument in favor of option #2 is the discussion about whether or not we want to use backdropcms.org for events OR if it is important for us to maintain that as it's own site. I am guessing, as indicated by @jenlampton that would could use one instance of CiviCRM for both functions even with two sites, but I suspect that will be additional work. I do not have strong feelings either way, but I suspect others might and I expect that this will require some discussion to work out. 

So, I would recommend that we do the following:

1) Keep working as quickly as possible to get CiviCRM working as part of BackdropCMS.org for contact management, donation management, etc. 
2) As soon as possible, have a larger discussion about how we want to integrate this with events. 
 - Do we want a stand-alone events site?
 - If we do have two different sites, how to integrate a single instance of CiviCRM into both.
3) Assume that events are being handled separately from the main instance until we have had the discussion in #2 and/or someone tells us that both are possible without slowing down the implementation of contact management.
